app/services/calculators/pensioner_capital_disregard_calculator.rb

```diff
@@ -1,51 +1,44 @@
 module Calculators
   class PensionerCapitalDisregardCalculator
-    def initialize(submission_date:, date_of_birth:, total_disposable_income:, receives_qualifying_benefit:)
-      @submission_date = submission_date
-      @date_of_birth = date_of_birth
-      @total_disposable_income = total_disposable_income
-      @receives_qualifying_benefit = receives_qualifying_benefit
-    end
-
-    def value
-      return 0 unless pensioner?
-
-      passported? ? passported_value : non_passported_value
-    end
-
-  private
+    class << self
+      def passported_value(submission_date:, date_of_birth:)
+        if pensioner?(submission_date:, date_of_birth:)
+          thresholds(submission_date)[:passported]
+        else
+          0
+        end
+      end
 
-    def pensioner?
-      if thresholds[:minimum_age_in_years] == "state_pension_age"
-        @submission_date > Calculators::StatePensionDateCalculator.state_pension_date(date_of_birth: @date_of_birth)
-      else
-        earliest_dob_for_pensioner >= @date_of_birth
+      def non_passported_value(submission_date:, date_of_birth:, total_disposable_income:)
+        if pensioner?(submission_date:, date_of_birth:)
+          income = total_disposable_income.to_f
+          thresholds(submission_date)[:monthly_income_values].detect { |value_bands, _banding| income_threshold_applies?(income, value_bands) }.second
+        else
+          0
+        end
       end
-    end
 
-    def earliest_dob_for_pensioner
-      @submission_date - thresholds[:minimum_age_in_years].years
-    end
+    private
 
-    def passported?
-      @receives_qualifying_benefit
-    end
-
-    def non_passported_value
-      income = @total_disposable_income.to_f
-      thresholds[:monthly_income_values].each { |value_bands, banding| return banding if income_threshold_applies(income, value_bands) }
-    end
+      def pensioner?(submission_date:, date_of_birth:)
+        if thresholds(submission_date)[:minimum_age_in_years] == "state_pension_age"
+          submission_date > Calculators::StatePensionDateCalculator.state_pension_date(date_of_birth:)
+        else
+          earliest_dob_for_pensioner(submission_date) >= date_of_birth
+        end
+      end
 
-    def passported_value
-      thresholds[:passported]
-    end
+      def earliest_dob_for_pensioner(submission_date)
+        submission_date - thresholds(submission_date)[:minimum_age_in_years].years
+      end
 
-    def income_threshold_applies(income, key_array)
-      (key_array.count.eql?(1) && income >= key_array[0]) || (income >= key_array[0] && income <= key_array[1])
-    end
+      def income_threshold_applies?(income, key_array)
+        (key_array.count.eql?(1) && income >= key_array[0]) || (income >= key_array[0] && income <= key_array[1])
```

Comment:
@starswan What do you think of using` income.between?(key_array[0], key_array[1])`?